SafeguardPasswords/test/Unit/TestRequest.query.pq

```diff
@@ -0,0 +1,47 @@
+section TestRequest;
+
+Request = Extension.ImportModule("Request.pqm");
+Request.GetAuthParameters = Request[GetAuthParameters];
+Request.GetDefaultParameters = Request[GetDefaultParameters];
+Request.GetSppResponse = Request[GetSppResponse];
+Request.ConvertQueryToRecord = Request[ConvertQueryToRecord];
+
+TestRequestGeneration = () =>
+    let
+
+        fakeresponse = FakeRawResponse([dummy_field = "value", dummy_list = {0, 1, 2}]),
+        expected_body = [dummy_field = "value", dummy_list = {0, 1, 2}],
+        expected_meta = [
+                            Content.Type = "application/json", 
+                            Response.Status = 200, 
+                            RequestUrl = "https://dummy_url", 
+                            RelativePath = "service/core/v4/fakePath"
+                        ],
+        response = Request.GetSppResponse("dummy_url", "fakeLoginToken", "service/core/v4/fakePath", null, "filter=fake ieq 'foo'&fields=fake,foo", [Response=fakeresponse]),
+
+        facts = {
+            Fact("Response body as json is correct", expected_body, response),
+            Fact("Response meta is correct", expected_meta, Value.Metadata(response))
+        }
+    in
+        facts;
+    
+TestQueryParsing = () =>
+    let
+        query = "filter=fake ieq 'foo'&fields=fake,foo",
+        expected = [filter="fake ieq 'foo'", fields="fake,foo"],
+        response = Request.ConvertQueryToRecord(query),
+
+        facts = {
+            Fact("Query parsed correctly", expected, response)
+        }
+    in
+        facts;
+
+shared TestRequest.UnitTest = [
+    facts = {
+        TestRequestGeneration(),
```

Comment:
Can you please add tests for these to: GetRstsResponse, GetLoginResponse, GetUnauthenticatedSppResponse, ConvertQueryToRecord?

Reply:
GetRstsResponse and GetLoginResponse are both tested in TestAuthentication.query.pq. While they are methods that perform an API request, they are part of the SPP RSTS authentication mechanism and I tested them in that context.

Ref https://github.com/OneIdentity/SafeguardPowerBI/pull/72/files#diff-4a787d39529b465af9b6fc4ec951e7798431291378f4bd84dc21fcc74b746e2e

I'll look at the other two methods.

Reply:
ACK.